readthedocs/core/unresolver.py

```diff
@@ -551,13 +551,23 @@ def unresolve_domain(self, domain):
         Unresolve domain by extracting relevant information from it.
 
         :param str domain: Domain to extract the information from.
+         It can be a full URL, in that case, only the domain is used.
         :returns: A UnresolvedDomain object.
         """
+        parsed_domain = urlparse(domain)
+        if parsed_domain.scheme:
+            if parsed_domain.scheme not in ["http", "https"]:
+                raise InvalidSchemeError(parsed_domain.scheme)
+            domain = parsed_domain.hostname
+
         public_domain = self.get_domain_from_host(settings.PUBLIC_DOMAIN)
         external_domain = self.get_domain_from_host(
             settings.RTD_EXTERNAL_VERSION_DOMAIN
         )
 
+        if not domain:
+            raise InvalidSubdomainError(domain)
+
```

Comment:
Do we not want to put this check on line 562, since it's checking for the domain set there?